ui/sdk/src/hamilton_sdk/api/clients.py

```diff
@@ -191,6 +192,9 @@ def __init__(
         ).start()
         self.worker_thread.start()
 
+        # Makes sure the process stops even if in remote environment
+        atexit.register(self.stop)
+
```

Comment:
this registers and ensures stop is called at process exit -- that's why that other code can be removed.